packages/custom-token-bridging/scripts/exec.js

```diff
@@ -0,0 +1,181 @@
+  console.log(
+    `Current retryable base submission prices for custom bridge and raouter are: ${_submissionPriceWeiForCustomBridge.toString(), _submissionPriceWeiForRouter.toString()}`
+  )
+  /**
+  * For the L2 gas price, we simply query it from the L2 provider, as we would when using L1
+  */
+  const gasPriceBid = await bridge.l2Provider.getGasPrice()
+  console.log(`L2 gas price: ${gasPriceBid.toString()}`)
+
+  /**
+  * For the gas limit, we'll simply use a hard-coded value (for more precise / dynamic estimates, see the estimateRetryableTicket method in the NodeInterface L2 "precompile")
+  */
+  const maxGasCustomBridge = 10000000
+  const maxGasRouter = 10000000
+
+
+  /**
+  * With these three values (base submission price, gas price, gas kinit), we can calculate the total callvalue we'll need our L1 transaction to send to L2
+  */
+  const valueForGateway = _submissionPriceWeiForCustomBridge.add(gasPriceBid.mul(maxGasCustomBridge))
+  const valueForRouter = _submissionPriceWeiForRouter.add(gasPriceBid.mul(maxGasRouter))
+  const callValue = valueForGateway.add(valueForRouter)
+
+  console.log(
+    `Registering the custom token on L2 with ${callValue.toString()} callValue for L2 fees:`
+  )
+  /** 
+    * Execute regsiterToken)nL2 function from our l1CustomToken
+    * Arguments include:
+    * l2CustomToken.address: The L2 address of your custom token
+    * _submissionPriceWeiForCustomBridge: Base submission price (in wei) that is needed to cover the cost for creating the retryable tickets for registering the token on custom gateway
+    * _submissionPriceWeiForRouter: Base submission price (in wei) that is needed to cover the cost for creating the retryable tickets for registering the token on gateway router
+    * maxGasCustomBridge: Gas limit for immediate L2 execution of registering the token on custom gateway
+    * maxGasRouter: Gas limit for immediate L2 execution of registering the token on router 
+    * gasPriceBid: The L2 gas price
+    * valueForGateway: the callvalue we need for our custom gateway
+    * valueForRouter: the callvalue we need for the router
+    */ 
+  const registerTokenTx = await l1CustomToken.registerTokenOnL2(
+    l2CustomToken.address,
+    _submissionPriceWeiForCustomBridge,
+    _submissionPriceWeiForRouter,
+    maxGasCustomBridge,
+    maxGasRouter,
+    gasPriceBid,
+    valueForGateway,
+    valueForRouter,
+    l2Wallet.address,
+    {
+      value: callValue
+    }
+  )
+
+  const registerTokenRec = await registerTokenTx.wait()
+  console.log(
+    `Registering token txn confirmed on L1! 🙌 ${registerTokenRec.transactionHash}`
+  )
+
+  /**
+   * The L1 side is confirmed; now we listen and wait for the for the Sequencer to include the L2 side; we can do this by computing the expected txn hash of the L2 transaction.
+   * To compute this txn hash, we need our message's "sequence number", a unique identifier. We'll fetch from the event logs with a helper method
```

Comment:
"we'll need our message's "sequence numbers", a unique identifiers of each L1 to L2 message"